client/src/pages/ProfileListings.js

```diff
@@ -66,40 +121,101 @@ function ProfileListings() {
   }
   if (sitters.data) {
     // set profile data to display
-    content = sitters.data.map((sitter) => (
-      <ProfileCard
-        key={sitter.profile._id}
-        firstName={sitter.profile.firstName}
-        lastName={sitter.profile.lastName}
-        profilePicture={sitter.profile.profilePicture}
-        address={sitter.profile.address}
-        rating={4}
-        price={sitter.profile.price}
-        description={sitter.profile.description}
-        userId={sitter._id}
-      />
-    ));
+    content = sitters.data
+      .filter(sitter => {
+        if (!filter || !filter.location) {
+          return sitter;
+        }
+        if (filter && filter.location) {
+          if (
+            sitter.profile.address &&
+            sitter.profile.address
+              .toLowerCase()
+              .includes(filter.location.toLowerCase().trim())
+          ) {
+            return sitter;
+          }
+        }
+      })
+      .filter(sitter => {
+        if (!filter || !filter.dropOff) {
+          return sitter;
+        }
+        if (filter && filter.dropOff) {
+          if (
+            sitter.profile.availability &&
+            sitter.profile.availability.length > 0
+          ) {
+            return sitter.profile.availability.find(timeSlot => {
+              if (
+                moment(filter.dropOff).format("YYMMDD") ===
+                  moment(timeSlot.start).format("YYMMDD") ||
+                moment(filter.dropOff).format("YYMMDD") ===
+                  moment(timeSlot.end).format("YYMMDD")
+              ) {
+                return sitter;
+              }
+            });
+          }
+        }
+      })
+      .filter(sitter => {
+        if (!filter || !filter.pickUp) {
+          return sitter;
+        }
+        if (filter && filter.pickUp) {
+          if (
+            sitter.profile.availability &&
+            sitter.profile.availability.length > 0
+          ) {
+            return sitter.profile.availability.find(timeSlot => {
+              if (
+                moment(filter.pickUp).format("YYMMDD") ===
+                  moment(timeSlot.end).format("YYMMDD") ||
+                moment(filter.pickUp).format("YYMMDD") ===
+                  moment(timeSlot.start).format("YYMMDD")
+              ) {
+                return sitter;
+              }
+            });
+          }
+        }
+      })
+      .map(sitter => (
+        <ProfileCard
+          key={sitter.profile._id}
+          firstName={sitter.profile.firstName}
+          lastName={sitter.profile.lastName}
+          profilePicture={sitter.profile.profilePicture}
+          address={sitter.profile.address}
+          rating={4}
+          price={sitter.profile.price}
+          description={sitter.profile.description}
+          userId={sitter._id}
+        />
+      ));
```

Comment:
See if you can refactor this into just one filter and one map

Reply:
Also I just noticed this, Array.filter expects a Boolean (or truthy/falsy value) to be returned in all cases so you will want to change that when you refactor

Reply:
Yes, you right, I saw those, but really forgot to address before PR

Reply:
Okay, regarding refactoring this into 1 filter, I believe it would not work that way. 
Here are possible approaches I see:

1)
`if(location match){
    if(dropoff match){
        if(pickup match){
            return sitter
        }
    }
}`
In this case we nest conditions, and if location wasn't chosen - we would never reach pick up date check

2)
`if(location match){
    //if we return sitter here, we would never check for dates. Otherwise we could stack sitter in some different array 
    //but in this case we would need to check that array against dates, update it if dates not match and then map it
    //looks overcomplicated to me
}
if(dropoff match){

}
if(pickup match){

}`
Here we check one by one and in my opinion this approach would complicate the logic